### PR TITLE
Feature/translate plugins

### PR DESCRIPTION
**Pull request by jonasheinrich**
https://community.openproject.org/work_packages/14922

### CODE REVIEW

File: lib/open_project/translations/models/locales_updater.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  def commit_and_push_plugin_repo(debug)
+    @plugin_repo.add('config/locales')
+    @plugin_repo.commit('update locales from crowdin')
+    @plugin_repo.push unless debug
+  end
+
+  def upload_english
+    # either add or update the english (js) translation file
+    titles = {
+      ENGLISH_TRANSLATION_FILE => 'OpenProject Wording',
```
Comment (ulferts) [inline lib/open_project/translations/models/locales_updater.rb]: @jonasheinrich can you put in the name of the plugin here? I think it would read nicer.

**Comment by ulferts** [comment]
@jonasheinrich we are out of time so I will merge this before it's time is due. However, there are some things that I require:

**Before merge**
- The specs fail and need to be fixed

**After merge as separate PR (unless possible before)**
- Documentation on how to create the Configuration.yml and how to run the plugin update. A configuration.yml.example would be nice.

File: lib/open_project/translations/models/locales_updater.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+require 'fileutils'
+
+require_relative '../helpers/tmp_directory'
+require_relative './git_repository'
+require_relative './i18n_provider'
+require_relative './locales_updater_configuration'
+
+ENGLISH_TRANSLATION_FILE = 'en.yml'
+ENGLISH_JS_TRANSLATION_FILE = 'js-en.yml'
+ACCEPTANCE_LEVEL = ENV['ACCEPTANCE_LEVEL'].nil? ? 100: ENV['ACCEPTANCE_LEVEL'].to_i
```
Comment (floriank) [inline lib/open_project/translations/models/locales_updater.rb:11]: While I get that this should mimic a constant (which do not really exist in ruby), this should probably be a method, as there is logic associated with it:

``` ruby
def acceptance_level
  return 100 if ENV['ACCEPTANCE_LEVEL'].nil?
  ENV['ACCEPTANCE_LEVEL'].to_i
end
```

Also - and this is just bitching - when I set the `ENV['ACCEPTANCE_LEVEL']` to something like `3e`, the level will be `0`.

At elast there is a space missing, if you want to keep it that way :wink:

File: lib/open_project/translations/models/locales_updater.rb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  def commit_and_push_plugin_repo(debug)
+    @plugin_repo.add('config/locales')
+    @plugin_repo.commit('update locales from crowdin')
+    @plugin_repo.push unless debug
+  end
+
+  def upload_english(plugin_name)
+    # either add or update the english (js) translation file
+    titles = {
+      ENGLISH_TRANSLATION_FILE => plugin_name +' Wording',
```
Comment (floriank) [inline lib/open_project/translations/models/locales_updater.rb:85]: `"#{plugin_name} Wording"`
?

**Comment by ulferts** [comment]
Sorry, @floriank, the page only updated after my merge

File: lib/open_project/translations/models/locales_updater_configuration.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require 'yaml'
+
+class LocalesUpdaterConfiguration
+  CONFIGURATION_FILE = ENV['CONFIGURATION_FILE'] || begin
```
Comment (floriank) [inline lib/open_project/translations/models/locales_updater_configuration.rb:4]: I wonder again why this is a constant - is it intended to be accessible from outside?

That would read something like

``` ruby
LocalesUpdaterConfiguration::CONFIGURATION_FILE
```

and that sounds a bit awkward. If it should be accessible from the outside, I'd suggest the name `FILE`..

I'd recommend putting this constant into a method (see above), as you could probably catch any errors on call.

**Comment by floriank** [comment]
No problem, I'd recommend a small refactoring before the release. Otherwise it will be put off to ... well, never.

**Pull request merged**